streams/src/test/java/org/apache/kafka/streams/kstream/TimeWindowedDeserializerTest.java

```diff
@@ -97,34 +104,41 @@ public void shouldThrowErrorIfWindowSizeSetInConstructorConfigAndConstructor() {
         assertThrows(IllegalArgumentException.class, () -> timeWindowedDeserializer.configure(props, false));
     }
 
+    @Deprecated
     @Test
     public void shouldThrowErrorIfWindowSizeIsNotSetAndWindowedInnerClassSerdeIsSet() {
         props.put(StreamsConfig.WINDOWED_INNER_CLASS_SERDE, Serdes.ByteArraySerde.class.getName());
-        final TimeWindowedDeserializer<?> deserializer = new TimeWindowedDeserializer<>();
-        assertThrows(IllegalArgumentException.class, () -> deserializer.configure(props, false));
+        try (final TimeWindowedDeserializer<?> deserializer = new TimeWindowedDeserializer<>()) {
+            assertThrows(IllegalArgumentException.class, () -> deserializer.configure(props, false));
+        }
     }
 
     @Test
     public void shouldThrowErrorIfWindowSizeIsNotSetAndWindowedInnerDeserializerClassIsSet() {
-        props.put(TimeWindowedDeserializer.WINDOW_SIZE_MS_CONFIG, Serdes.ByteArraySerde.class.getName());
-        final TimeWindowedDeserializer<?> deserializer = new TimeWindowedDeserializer<>();
-        assertThrows(IllegalArgumentException.class, () -> deserializer.configure(props, false));
+        props.put(TimeWindowedDeserializer.WINDOWED_INNER_DESERIALIZER_CLASS, Serdes.ByteArraySerde.class.getName());
```

Comment:
This is the actually bug-fix -- the test used `WINDOW_SIZE_MS_CONFIG` incorrectly, however, this lead to a `NumberFormatException` which extends `IllegalArgumentException` and to the test still passed...

Reply:
How about using `assertThrowsExactly` to ensure that only the expected exception type is thrown, without allowing subclasses?

Reply:
Interesting idea. For this PR, I think it would just be more "busy work". But in general, it could be a reasonable idea.